src/main/kotlin/de/evoila/osb/checker/tests/containers/BindingContainers.kt

```diff
@@ -118,21 +120,37 @@ class BindingContainers(
                             expectedResponseBody = NO_SCHEMA
                     )
                 },
-                DynamicTest.dynamicTest("Running PUT binding with different attribute again. Expecting StatusCode 409.") {
-                    bindingRequestRunner.runPutBindingRequestAsync(
-                            requestBody = binding.copy(
-                                    parameters = binding.parameters?.plus(Pair("key_osb-checker-kotlin", "additional-parameter"))
-                                            ?: mapOf(Pair("key_osb-checker-kotlin", "parameter"))
-                            ),
-                            instanceId = instanceId,
-                            bindingId = bindingId,
-                            expectedStatusCodes = *intArrayOf(409),
-                            expectedResponseBody = NO_SCHEMA
-                    )
+                if (catalogHasMultiplePlans()) {
+                    DynamicTest.dynamicTest("Running PUT binding with different service or plan id, but same instance and binding id again. Expecting StatusCode 409.") {
+                        bindingRequestRunner.runPutBindingRequestAsync(
+                                requestBody = swapServiceAndPlanId(binding),
+                                instanceId = instanceId,
+                                bindingId = bindingId,
+                                expectedStatusCodes = *intArrayOf(409),
+                                expectedResponseBody = NO_SCHEMA
+                        )
+                    }
+                } else {
+                    DynamicTest.dynamicTest("%SKIPPED%Skipping PUT Binding with different attributes, because catalog does not contain more than one plan.") {}
                 }
         )
     }
 
+    private fun catalogHasMultiplePlans(): Boolean = catalogRequestRunner.correctRequest().services.flatMap { it.plans }.size > 1
+
+    private fun swapServiceAndPlanId(oldBindingBody: BindingBody): BindingBody {
+        val catalog = catalogRequestRunner.correctRequest()
+        return if (catalog.services.size > 1) {
+            catalog.services.first { service -> service.id != oldBindingBody.serviceId }.let {
+
+                oldBindingBody.copy(serviceId = it.id, planId = it.plans.first().id)
+            }
+        } else {
```

Comment:
Would have switch the case prioritization, so that you use a different plan of the service first before using another service definition.